### PR TITLE
Pull in old PRs from Chris's repo

### PR DESCRIPTION
**Pull request by danhunsaker**
This PR consists entirely of changes submitted by PHP Resque users over the last several years. A small handful of PRs remain unpulled (~~chrisboulton/php-resque#35~~, chrisboulton/php-resque#81, chrisboulton/php-resque#338), but they're complex enough to deserve their own, separate PRs to this repo, so I've mentioned that to their creators (in the original PRs), and hopefully they'll bring them over here.

### CODE REVIEW

File: lib/Resque/Worker.php [hunk 1]
```diff
@@ -217,7 +257,14 @@ public function work($interval = Resque::DEFAULT_INTERVAL, $blocking = false)
 				$this->logger->log(Psr\Log\LogLevel::INFO, $status);
 
 				// Wait until the child process finishes before continuing
-				pcntl_wait($status);
+				while (pcntl_wait($status, WNOHANG) === 0) {
```
[the remark this answers is not in the document]
Comment (danhunsaker) [reply lib/Resque/Worker.php]: I'd say pull this branch down and give it a test. See if the current fix is sufficient. If not, toss us a PR with your changes as well, and we'll add them, too!

[the remark this answers is not in the document]
Comment (Micr0mega) [reply lib/Resque/Worker.php]: I have done so and can confirm this change is working as expected, so no need for additional changes!

**Comment by Micr0mega** [comment]
So what is the plan for having these reviewed? Can't wait for a new release ;-)

**Comment by danhunsaker** [comment]
Just waiting for somebody on the PHP Resque team (@resque/php) who isn't myself to submit a review. Can't review my own PRs! (GitHub literally doesn't allow it...) Once that's done, merging will be pretty quick!

**Comment by danhunsaker** [comment]
In the absence of reviews from the @resque/php team, I'm overriding the requirement, as a repo admin, for a review, and merging this now. It will go to the `develop` branch, not the `master` branch, so folks will have a chance to test it properly (gonna tag it with a `-beta` flag, so you need to enable `@dev` stability for this) before it gets tagged as a stable release. Please submit feedback here!

**Comment by Micr0mega** [comment]
Deployed this two days ago where it would have little impact if broken. It has been running at least a job per minute, not a single problem so far :+1:

On different note, when I follow the link in the @resque/php mentions, I get a 404. Maybe this is related to nobody responding to the call for review?

**Comment by Micr0mega** [comment]
> You mean the link to the team itself? Or something else?

The former, when referencing the team in your previous comments here. For example:
> In the absence of reviews from the @resque/php team, I'm overriding the requirement, as a repo admin, for a review, and merging this now. It will go to the `develop` branch, not the `master` branch, so folks will have a chance to test it properly (gonna tag it with a `-beta` flag, so you need to enable `@dev` stability for this) before it gets tagged as a stable release. Please submit feedback here!

@resque/php links to https://github.com/orgs/resque/teams/php, but that's a 404 for me. So I was thinking they might not have received any notification about your request to review, because something is wrong with the team. Might be completely wrong though :) I cannot seem to reproduce the reference in a comment either, so I don't know how it works.

**Comment by danhunsaker** [comment]
Ah. Must mean the team is private. Interesting they've opted to 404 those rather than 403, but ¯\_(ツ)_/¯ I guess.

**Pull request merged**